src/npcmove.cpp

```diff
@@ -931,31 +909,91 @@ npc_action npc::address_needs()
     return address_needs( ai_cache.danger );
 }
 
-bool npc::can_reload()
+bool wants_to_reload( const item &it )
 {
-    return find_usable_ammo();
+    if( !it.can_reload() ) {
+        return false;
+    }
+
+    const int required = it.ammo_required();
```

Comment:
consider `item::is_reloadable()` and/or `item::can_reload()` as these will be updated when bandoliers become directly reloadable.